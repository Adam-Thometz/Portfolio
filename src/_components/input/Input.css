.Input {
  position: relative;
  border: none;
}

.Input label {
  position: absolute;
  top: 16px;
  left: 0;
  font-size: 16px;
  transition: .5s ;
  padding-left: 16px;
  color: #CACACA;
}

.Input input, .Input textarea {
  width: 100%;
  height: 50px;
  /* padding: 24px 16px 4px 16px; */
  color: white;
  margin-bottom: 30px;
  background-color: #171717;
  box-sizing: border-box;
  box-shadow: none;
  outline: none;
  border: none;
  border-bottom: 1.5px solid #C2C2C2;
  font-size: 1.2rem;
  border-radius: 4px 4px 0 0;
}

.Input textarea {
  width: 100%;
  height: 100px;
  resize: none;
}

.Input input:focus, .Input textarea:focus {
  border-bottom: 1.5px solid #50BAC7;
  background-color: #747373;
}

.Input input:focus ~ label,
.Input input:valid ~ label,
.Input textarea:focus ~ label,
.Input textarea:valid ~ label {
  top: 8px;
  font-size: 0.8rem;
}