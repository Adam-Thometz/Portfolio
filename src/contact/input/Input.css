.Input {
  position: relative;
  border: none;
  padding: 0;
  margin-bottom: 30px;
  width: 50%;
}

.Input label {
  position: absolute;
  top: 12px;
  left: 0;
  font-size: 16px;
  transition: 0.25s;
  padding-left: 16px;
  color: #cacaca;
}

.Input input,
.Input textarea {
  width: 100%;
  height: 50px;
  padding: 24px 16px 4px 16px;
  color: white;
  background-color: #17171766;
  box-sizing: border-box;
  box-shadow: none;
  outline: none;
  border: none;
  font-family: "Roboto", sans-serif;
  border-bottom: 1.5px solid #c2c2c2;
  font-size: 1.2rem;
  border-radius: 4px;
}

.Input textarea {
  height: 100px;
  width: 199%;
}

.Input input:focus,
.Input textarea:focus {
  border-bottom: 1.5px solid #50bac7;
  background-color: #74737366;
}

.Input input:focus ~ label,
.Input input:valid ~ label,
.Input textarea:focus ~ label,
.Input textarea:valid ~ label {
  top: 8px;
  font-size: 0.8rem;
}

@media (max-width: 500px) {
  .Input input {
    width: 100%;
  }
}
