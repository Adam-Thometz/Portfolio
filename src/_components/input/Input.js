import React from 'react';

import './Input.css'

const Input = ({ label, name, id, type = 'text', value, onChange }) => {
  return (
    <fieldset className='Input'>
      <label htmlFor={name}>{label}</label>
      {type === 'textarea' ? 
      <textarea 
        name={name}
        id={id}
        onChange={onChange}
        required
      >{value}</textarea>
      : <input 
        type={type}
        name={name}
        id={id}
        value={value}
        onChange={onChange}
        required
      />}
    </fieldset>
  );
};

export default Input;