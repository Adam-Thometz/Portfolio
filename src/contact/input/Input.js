import React from 'react';

import './Input.css'

const Input = ({ label, name, id, type = 'text', value, onChange }) => {
  return (
    <fieldset className='Input'>
      {type === 'textarea' ? 
        <textarea 
          name={name}
          id={id}
          onChange={onChange}
          required
          defaultValue={value}
        ></textarea>
      : <input 
        type={type}
        name={name}
        id={id}
        value={value}
        onChange={onChange}
        required
      />}
      <label htmlFor={name}>{label}</label>
    </fieldset>
  );
};

export default Input;