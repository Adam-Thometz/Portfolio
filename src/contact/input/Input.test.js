import React from "react";
import { render } from "@testing-library/react";
import Input from './Input';

describe('Input component', () => {
  it('renders without crashing', () => {
    render(<Input />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});