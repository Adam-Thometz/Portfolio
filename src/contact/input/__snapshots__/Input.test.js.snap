// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches the snapshot 1`] = `
<DocumentFragment>
  <fieldset
    class="Input"
  >
    <input
      required=""
      type="text"
      value=""
    />
    <label />
  </fieldset>
</DocumentFragment>
`;
